### ISSUE TITLE
Document src/transpile/fusion.hpp

### ISSUE THREAD

**Issue by yaelbh** [Qiskit/qiskit-aer#607]

**Comment by hhorii** [comment]
I'm now refactoring this class to use unitarymatrix simulator in `make_fusion`. I will send a PR after #606 with more comments.

**Comment by chriseclectic** [comment]
@hhorii Lets make a separate PR for improving gate fusion. I think your PR 606 closes the documentation issue from the QasmSimulator class doc, but maybe some more documentation in the source code would be helpful as well.
